### PR TITLE
pull request test

### PR DESCRIPTION
**Pull request by ibailisitski**

### CODE REVIEW

File: first.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+// 1
+
+let numberSetOne = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+function modifyNumbers(numbers) {
+    let modifyNumberSet = [];
```
Comment (Boobaley) [inline first.js]: use `const` instead. we use `let` only when we need to override variable.

File: first.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    for (let i = 0; i < numbers.length; i++) {
+        if (numbers[i] % 2 === 0) {
+            modifyNumberSet.push(numbers[i] += 1);
+        } else {
+            modifyNumberSet.push(numbers[i] -= 1);
```
Comment (Boobaley) [inline first.js]: you use `numbers[i]` 2 times. better to extract this into variable and then use.

`const currentNumber = numbers[i]`

File: first.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        }
```
Comment (Boobaley) [inline first.js]: we can avoid this nested `if else` structure by this twik:

```javascript
const numberToPush = currentNumber % 2 === 0 ? currentNumber + 1 : currentNumber -1;
modifyNumberSet.push(numberToPush )
```

File: first.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    return modifyNumberSet;
+}
+
+console.log(modifyNumbers(numberSetOne));
+
+
+// 2
+
+function splitWordsIntoLetters(words) {
+    let wordIntoLetterSet = [];
```
Comment (Boobaley) [inline first.js]: try to avoid `let` usages. 90% cases you need to use `const`

File: first.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+function modifyNumbers(numbers) {
+    const modifyNumberSet = [];
+
+    for (let i = 0; i < numbers.length; i++) {
+        // if (numbers[i] % 2 === 0) {
+        //     modifyNumberSet.push(numbers[i] += 1);
+        // } else {
+        //     modifyNumberSet.push(numbers[i] -= 1);
+        // }
+        let currentNumber = numbers[i];
```
Comment (Boobaley) [inline first.js:16]: why `let`?

File: first.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        modifyNumberSet.push(currentNumber % 2 === 0 ? currentNumber += 1 : currentNumber -= 1);
```
Comment (Boobaley) [inline first.js:17]: `currentNumber % 2 === 0 ? currentNumber + 1 : currentNumber - 1`

File: first.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+        firstLetter = firstLetter.toUpperCase();
+
+        let remainingLetters = word.slice(1);
+        remainingLetters = remainingLetters.toLowerCase();
+
+        let capitalizedWord = firstLetter + remainingLetters;
+
+        return capitalizedWord;
+    });
+}
```
Comment (Boobaley) [inline first.js:41]:
```javascript
const firstLetter = word[0].toUpperCase();
const restLetters = word.slice(1).toLowerCase()

return `${firstLetter}${restLetters}` 
``` 
For more details check this out: https://learn.javascript.ru/types#stroka
looks shorter and mach cleaner

File: first.js [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    {
+        id: 4,
+        name: "Four",
+    }
+];
+
+function removeUsers(usersArray, idToRemove) {
+    const filteredUsers = usersArray.filter(user => !idToRemove.includes(user.id));
+    return filteredUsers; 
+}
```
Comment (Boobaley) [inline first.js:72]:
```javascript
function removeUsers(usersArray, idToRemove) {
   return usersArray.filter(user => !idToRemove.includes(user.id));
}
```

File: first.js [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+const idToRemove = [1, 3];
+console.log(removeUsers(users, idToRemove));
+
+
+// 4 ---------------
+
+function isPalindrome(word) {
+    const wordReverse = word.split('').reverse().join('');
+    return wordReverse === word;
+};
```
Comment (Boobaley) [inline first.js:83]: good, did you google it ? :)

**Review by Boobaley — CHANGES_REQUESTED** [review]

File: first.js [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+    for (let i = 0; i < numbers.length; i++) {
+        // if (numbers[i] % 2 === 0) {
+        //     modifyNumberSet.push(numbers[i] += 1);
+        // } else {
+        //     modifyNumberSet.push(numbers[i] -= 1);
+        // }
+        let currentNumber = numbers[i];
+        modifyNumberSet.push(currentNumber % 2 === 0 ? currentNumber += 1 : currentNumber -= 1);
+
```
Comment (Boobaley) [inline first.js:18]: rework this task with `map` method

File: first.js (hunk 5 above)
Comment (ibailisitski) [reply first.js:16]: this is a changeable variable, and it doesn't work with `const`

**Pull request merged**